lib/validator.js

```diff
@@ -72,14 +72,28 @@ module.exports = function(){
 
     /**
      * Validates a token from an Express request object
-     * 
+     *
      * @param  {Express request}   req the Express request object
-     * @param  {Function} cb  Callback when to be called when token 
+     * @param  {Function} cb  Callback when to be called when token
      *                        has been validated or an error has occured
      */
     validateTokenRequest: function(req, cb){
       var self = this;
-      var token = waterlock._utils.allParams(req).access_token;
+      var token = null;
+      console.log(req.headers);
+      if (req.headers && req.headers.authorization) {
+          var parts = req.headers.authorization.split(' ');
+          if (parts.length === 2){
+            var scheme = parts[0];
+            var credentials = parts[1];
+
+            if (/^Bearer$/i.test(scheme)){
+              token = credentials;
+            }
+          }
+      }else{
+          token = waterlock._utils.allParams(req).access_token;
+      }
```

Comment:
Is there a way we can contain the logic from lines 84-96 in a different private function in this file? The method validateTokenRequest is long enough already so it will only get harder and harder to maintain as time goes on. If we put this logic for the bearer in a different function and call that function in here then if the logic for this check ever changes or we need to remove it, it will be a very trivial change, it will also be easier to test. Nice work though keep it up!